From b0da9dd969384c91e3a6abe41e88dbe9d441f92c Mon Sep 17 00:00:00 2001
From: CI Assistant <yossi.ilkanaev@intel.com>
Date: Mon, 26 Nov 2018 13:09:01 +0100
Subject: [PATCH] Merge pull request #291 in SW_UGW/linux from
 bugfix/DRVLIB_SW-961-Adjust-DMA-data-offset-align-128bytes to master

* commit '738014984fc59c071b3b23e6ddd759617d7cf8c6':
  Reduce the skb headroom and align the DMA data offset 128 Bytes
---
 drivers/net/ethernet/lantiq/cqm/cqm_common.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/cqm_common.h b/drivers/net/ethernet/lantiq/cqm/cqm_common.h
index ab41357c5904..549d360061da 100644
--- a/drivers/net/ethernet/lantiq/cqm/cqm_common.h
+++ b/drivers/net/ethernet/lantiq/cqm/cqm_common.h
@@ -13,7 +13,11 @@
 #define LS_PORT_NUM             4
 #define CBM_MAX_INT_LINES       8
 #define CBM_DMA_DESC_OWN        1U/*belong to DMA*/
-#define CBM_DMA_DATA_OFFSET     128
+ /* DMA offset to be 128 byte aligned
+  * CBM_DMA_DATA_OFFSET + NET_IP_ALIGN + NET_SKB_PAD +
+  * CQM_POOL_METADATA = 128 + 6(byte offset for IP ALIGN)
+  */
+#define CBM_DMA_DATA_OFFSET     96
 /* TCP lite and LRO expects it to be 128 */
 #define CBM_GRX550_DMA_DATA_OFFSET     128
 #define DEFAULT_WAIT_CYCLES     20
